client/me/developer/main.jsx

```diff
@@ -117,46 +141,74 @@ class Developer extends Component {
+			this.showDeveloperSurveyNotice();
 		}
 	}
 
 	render() {
 		return (
-			<Main className="developer" wideLayout>
-				<PageViewTracker path="/me/developer" title="Me > Developer" />
-				<ReauthRequired twoStepAuthorization={ twoStepAuthorization } />
-				<NavigationHeader
-					navigationItems={ [] }
-					title={ this.props.translate( 'Developer Features' ) }
-					subtitle={ this.props.translate(
-						'Take WordPress.com further with early access to new developer features.'
+			<>
+				<Main className="developer" wideLayout>
+					<PageViewTracker path="/me/developer" title="Me > Developer" />
+					<ReauthRequired twoStepAuthorization={ twoStepAuthorization } />
+					<NavigationHeader
+						navigationItems={ [] }
+						title={ this.props.translate( 'Developer Features' ) }
+						subtitle={ this.props.translate(
+							'Take WordPress.com further with early access to new developer features.'
+						) }
+						className="developer__header"
+					/>
+
+					<form onChange={ this.props.submitForm }>
+						<FormFieldset
+							className={ classnames( 'developer__is_dev_account-fieldset', {
+								'is-loading': this.props.isFetchingUserSettings,
+							} ) }
+						>
+							<ToggleControl
+								disabled={ this.props.isFetchingUserSettings || this.props.isUpdatingUserSettings }
+								checked={ this.props.getSetting( 'is_dev_account' ) }
+								onChange={ this.handleToggleIsDevAccount }
+								label={ getIAmDeveloperCopy( this.props.translate ) }
+							/>
+						</FormFieldset>
+					</form>
+
+					<DeveloperFeatures />
+				</Main>
+
+				{ this.state.isDeveloperSurveyNoticeVisible &&
+					ReactDOM.createPortal(
+						<DeveloperSurveyNotice
+							localeSlug={ this.props.currentUser.localeSlug }
+							onSurveyClick={ () => {
+								recordTracksEvent( 'calypso_me_developer_survey_clicked' );
+
+								this.hideDeveloperSurveyNotice();
+							} }
+							onClose={ ( remindTimeInSeconds, buttonName ) => {
+								recordTracksEvent( 'calypso_me_developer_survey_dismissed', {
+									context: buttonName,
+								} );
+
+								this.setDeveloperSurveyCookie( 'dismissed', remindTimeInSeconds );
+
+								this.hideDeveloperSurveyNotice();
+							} }
+						/>,
+						this.surveyNoticeWrapper
 					) }
-					className="developer__header"
-				/>
-
-				<form onChange={ this.props.submitForm }>
-					<FormFieldset
-						className={ classnames( 'developer__is_dev_account-fieldset', {
-							'is-loading': this.props.isFetchingUserSettings,
-						} ) }
-					>
-						<ToggleControl
-							disabled={ this.props.isFetchingUserSettings || this.props.isUpdatingUserSettings }
-							checked={ this.props.getSetting( 'is_dev_account' ) }
-							onChange={ this.handleToggleIsDevAccount }
-							label={ getIAmDeveloperCopy( this.props.translate ) }
-						/>
-					</FormFieldset>
-				</form>
-
-				<DeveloperFeatures />
-			</Main>
+			</>
```

Comment:
```tsx
{ this.state.isDeveloperSurveyNoticeVisible && (
	<DeveloperSurveyNotice
		localeSlug={ this.props.currentUser.localeSlug }
		onSurveyClick={ () => {
			recordTracksEvent( 'calypso_me_developer_survey_clicked' );

			this.hideDeveloperSurveyNotice();
		} }
		onClose={ ( remindTimeInSeconds, buttonName ) => {
			recordTracksEvent( 'calypso_me_developer_survey_dismissed', {
				context: buttonName,
			} );

			this.setDeveloperSurveyCookie( 'dismissed', remindTimeInSeconds );

			this.hideDeveloperSurveyNotice();
		} }
	/>
) }
```

On the one hand, not a super big deal, but on the other hand, I don't see the need for using a React portal here. The code snippet I included seems to yield the same result.

@nightnei, was there a particular reason for using a React portal here?

Reply:
My reason is reliable UI - there is chance to have some `position: relative` or `z-index` or similar thing in chain of parents, to break covering the whole UI with backdrop. I faced such cases in my experience. Portals are created specifically to avoid such issues.
Also, it's just personal, but I feel semantically more right to put such notices to the `<body>` since it has the same "level" as all content. Under "level" I mean - it can compete with the whole content as more prior.

Reply:
Sure sure, that makes sense 👍 In this particular instance, though, `Developer` renders at such a high level in the component tree that the risk seems low.

But I'm still not advocating that we make any change here – that seems like unnecessary work